internal/service/arcregionswitch/plan.go

```diff
@@ -0,0 +1,1373 @@
+	TimeoutMinutes           types.Int64  `tfsdk:"timeout_minutes"`
+	RegionAndRoutingControls types.List   `tfsdk:"region_and_routing_controls"`
+}
+
+type regionAndRoutingControlsModel struct {
+	Region             types.String `tfsdk:"region"`
+	RoutingControlArns types.List   `tfsdk:"routing_control_arns"`
+}
+
+// Parallel Configuration Models
+type parallelConfigModel struct {
+	Step types.List `tfsdk:"step"`
+}
+
+type parallelStepModel struct {
+	Name                     types.String `tfsdk:"name"`
+	ExecutionBlockType       types.String `tfsdk:"execution_block_type"`
+	Description              types.String `tfsdk:"description"`
+	ExecutionApprovalConfig  types.List   `tfsdk:"execution_approval_config"`
+	CustomActionLambdaConfig types.List   `tfsdk:"custom_action_lambda_config"`
+}
+
+func (r *resourcePlan) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan resourcePlanModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().ARCRegionSwitchClient(ctx)
+
+	var regions []string
+	resp.Diagnostics.Append(plan.Regions.ElementsAs(ctx, &regions, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	input := arcregionswitch.CreatePlanInput{
+		Name:             aws.String(plan.Name.ValueString()),
+		ExecutionRole:    aws.String(plan.ExecutionRole.ValueString()),
+		RecoveryApproach: awstypes.RecoveryApproach(plan.RecoveryApproach.ValueString()),
+		Regions:          regions,
+	}
+
+	if !plan.Description.IsNull() {
+		input.Description = aws.String(plan.Description.ValueString())
+	}
+
+	if !plan.PrimaryRegion.IsNull() {
+		input.PrimaryRegion = aws.String(plan.PrimaryRegion.ValueString())
+	}
+
+	if !plan.RecoveryTimeObjectiveMinutes.IsNull() {
+		input.RecoveryTimeObjectiveMinutes = aws.Int32(int32(plan.RecoveryTimeObjectiveMinutes.ValueInt64()))
+	}
+
+	// Handle workflows - API requires this field
+	if !plan.Workflow.IsNull() {
+		var workflows []workflowModel
+		resp.Diagnostics.Append(plan.Workflow.ElementsAs(ctx, &workflows, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		input.Workflows = expandWorkflowsFromFramework(workflows)
+	}
+
+	// Handle associated alarms
+	if !plan.AssociatedAlarms.IsNull() && !plan.AssociatedAlarms.IsUnknown() {
+		var alarms []associatedAlarmModel
+		resp.Diagnostics.Append(plan.AssociatedAlarms.ElementsAs(ctx, &alarms, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		input.AssociatedAlarms = expandAssociatedAlarmsFromFramework(alarms)
+	}
+
+	// Handle tags - use getTagsIn to get all tags including provider defaults
+	input.Tags = getTagsIn(ctx)
+
+	output, err := conn.CreatePlan(ctx, &input)
+	if err != nil {
+		resp.Diagnostics.AddError("creating ARC Region Switch Plan", err.Error())
+		return
+	}
+
+	plan.ARN = types.StringValue(aws.ToString(output.Plan.Arn))
+	plan.ID = types.StringValue(aws.ToString(output.Plan.Arn))
```

Comment:
Since the `id` attribute is redundant here, we can drop it and use `arn` as the primary identifier.

https://hashicorp.github.io/terraform-provider-aws/id-attributes/#standard-usage